### ISSUE TITLE
Update document to reflect what pull/2410 improved block_in_place()

### ISSUE THREAD

**Issue by qiujiangkun** [tokio-rs/tokio#3776]
As of 1.5.0, documentation about `block_in_place` implies that it should only be used inside a tokio runtime. However, #2410 expanded `block_in_place` usage but did not touch the documentation, which should be changed as well.

```text
Runs the provided blocking function on the current thread without
blocking the executor.
In general, issuing a blocking call or performing a lot of compute in a
future without yielding is not okay, as it may prevent the executor from
driving other futures forward.  This function runs the closure on the
current thread by having the thread temporarily cease from being a core
thread, and turns it into a blocking thread. See the [CPU-bound tasks
and blocking code][blocking] section for more information.
Although this function avoids starving other independently spawned
tasks, any other code running concurrently in the same task will be
suspended during the call to `block_in_place`. This can happen e.g. when
using the [`join!`] macro. To avoid this issue, use [`spawn_blocking`]
instead.
Note that this function can only be used when using the `multi_thread` runtime.
Code running behind `block_in_place` cannot be cancelled. When you shut
down the executor, it will wait indefinitely for all blocking operations
to finish. You can use [`shutdown_timeout`] to stop waiting for them
after a certain timeout. Be aware that this will still not cancel the
tasks — they are simply allowed to keep running after the method
returns.
```

**Comment by Darksonn** [comment]
This was already fixed by the not-yet released PR #3753.

**Comment by taiki-e** [comment]
Closing -- tokio 1.6.0 released
